book_maker/translator/chatgptapi_translator.py

```
@@ -134,27 +157,45 @@ def create_context_messages(self):
+            cur_content = re.sub(r'<think>.*?</think>','',cur_content,flags=re.S) #r1-like things
+            if cur_content is not None:
+                t_text += cur_content.encode("utf8").decode() or ""
+            else:
+                break
+            if completion.choices[0].finish_reason != "length":
+                break
+            if needprint:
+                _comp_len_info = f"completion_tokens: {completion.usage.completion_tokens}" if completion.usage.completion_tokens else f"len(completion): {len(cur_content)}"
+                print(f"[bold red]Imcompleted translation due to length at Attempt {len_retry+1}; {_comp_len_info}[/bold red]")
+            if len(text) * 3 < len(t_text):
+                print(f"[bold red]Length limit exceeded and output seems too long[/bold red]")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)